test/e2e/i18n-api-support/index.test.ts

```diff
@@ -0,0 +1,71 @@
+import { createNext } from 'e2e-utils'
+import { fetchViaHTTP } from 'next-test-utils'
+import { NextInstance } from 'test/lib/next-modes/base'
+
+describe('i18n API support', () => {
+  let next: NextInstance
+
+  beforeAll(async () => {
+    next = await createNext({
+      files: {
+        'pages/api/hello.js': `
+          export default function handler(req, res) { 
+            res.end('hello world')
+          } 
+        `,
+        'pages/api/blog/[slug].js': `
+          export default function handler(req, res) {
+            res.end('blog/[slug]')
+          }
+        `,
+      },
+      nextConfig: {
+        i18n: {
+          locales: ['en', 'fr'],
+          defaultLocale: 'en',
+        },
+        async rewrites() {
+          return {
+            beforeFiles: [],
+            afterFiles: [],
+            fallback: [
+              {
+                source: '/api/:path*',
+                destination: 'https://example.com/',
```

Comment:
Let’s use example.vercel.sh instead?

Reply:
Sounds good, updated in https://github.com/vercel/next.js/pull/33966/commits/bdfe2cb3942ce6cc8666f76074183c6c46e4dd90